bgpd/bgp_linkstate_tlv.c

```diff
@@ -388,6 +446,46 @@ static int bgp_linkstate_nlri_link_descriptor_decode(
 	return BGP_NLRI_PARSE_OK;
 }
 
+static void bgp_linkstate_nlri_link_descriptor_encode(
+	struct bgp_ls_nlri_link_descr_tlv *link_descr, struct stream *s)
+{
+	uint16_t *pnt;
+
+	if (CHECK_FLAG(link_descr->flags,
+		       BGP_NLRI_TLV_LINK_DESCR_LOCAL_REMOTE_ID)) {
+		stream_putw(s, BGP_LS_TLV_LINK_LOCAL_REMOTE_IDENTIFIERS);
+		stream_putw(s, 2);
+		stream_putw(s, link_descr->local_remote_id);
+	}
+	if (CHECK_FLAG(link_descr->flags, BGP_NLRI_TLV_LINK_DESCR_INTERFACE4)) {
+		stream_putw(s, BGP_LS_TLV_IPV4_INTERFACE_ADDRESS);
+		stream_putw(s, IPV4_MAX_BYTELEN);
+		stream_put_in_addr(s, &link_descr->interface4);
+	}
+	if (CHECK_FLAG(link_descr->flags, BGP_NLRI_TLV_LINK_DESCR_NEIGHBOR4)) {
+		stream_putw(s, BGP_LS_TLV_IPV4_NEIGHBOR_ADDRESS);
+		stream_putw(s, IPV4_MAX_BYTELEN);
+		stream_put_in_addr(s, &link_descr->neighbor4);
+	}
+	if (CHECK_FLAG(link_descr->flags, BGP_NLRI_TLV_LINK_DESCR_INTERFACE6)) {
+		stream_putw(s, BGP_LS_TLV_IPV6_INTERFACE_ADDRESS);
+		stream_putw(s, IPV6_MAX_BYTELEN);
+		stream_put(s, &link_descr->interface6, IPV6_MAX_BYTELEN);
+	}
+	if (CHECK_FLAG(link_descr->flags, BGP_NLRI_TLV_LINK_DESCR_NEIGHBOR6)) {
+		stream_putw(s, BGP_LS_TLV_IPV6_NEIGHBOR_ADDRESS);
+		stream_putw(s, IPV6_MAX_BYTELEN);
+		stream_put(s, &link_descr->neighbor6, IPV6_MAX_BYTELEN);
+	}
+	if (CHECK_FLAG(link_descr->flags, BGP_NLRI_TLV_LINK_DESCR_MT_ID)) {
+		stream_putw(s, BGP_LS_TLV_MULTI_TOPOLOGY_ID);
+		stream_putw(s, link_descr->mtid.length);
+		pnt = (uint16_t *)&link_descr->mtid.data;
+		for (int i = 0; i < (link_descr->mtid.length / 2); i++)
+			stream_putw(s, pnt[i]);
+	}
+}
```

Comment:
unknown node TLVs should somehow be encoded.

Reply:
see https://github.com/FRRouting/frr/pull/12649#discussion_r1169925019